share/functions/__fish_inc_dec_number_under_cursor.fish

```diff
@@ -0,0 +1,90 @@
+# Usage:
+    # This function can be bound to key shortcuts (like Ctrl+A for increase and Ctrl+X for decrease) in fish shell to modify numbers quickly under the cursor.
+    # Key bindings for increasing and decreasing numbers in insert/vi mode mode.
+    # bind -M insert \ca '__fish_inc_dec_number_under_cursor increase'
+    # bind -M insert \cx '__fish_inc_dec_number_under_cursor decrease'
+
+function __fish_inc_dec_number_under_cursor
+    # Capture the current command line buffer and cursor position.
+    set current_buffer (commandline -b)  # Get the entire current command line as a string.
```

Comment:
use `set -l` or `set -f` to avoid writing to global/universal variables. Same below